bootstrap/090_post_setup.sql

```diff
@@ -69,80 +69,109 @@ CREATE OR REPLACE TASK TASKS.PROBE_MONITORING
+                EXCEPTION
+                    when EXPRESSION_ERROR THEN
+                        outcome2 := SQLERRM;
+                    WHEN other THEN
+                        outcome2 := SQLERRM;
+                END;
+                outcome := outcome || outcome2;
+            end if;
+
+            -- Send slack messages
+            let slackDests string := act.action_taken:SLACK;
+            IF (length(slackDests) > 1) THEN
+                let slackResult string := '';
+                let dict variant := OBJECT_CONSTRUCT('bt', CHAR(UNICODE('\u0060')), 'query_id', act.query_id, 'query_text', act.query_text, 'user_name', act.user_name, 'warehouse_name', act.warehouse_name, 'start_time', act.start_time, 'probe_name', act.probe_name);
+                let message string := '
+    Sundeck OpsCenter probe [{probe_name}] matched query.\n
+    Query Id: {bt}{query_id}{bt}\n
+    Query User: {user_name}\n
+    Warehouse Name: {bt}{warehouse_name}{bt}\n
+    Start Time: {bt}{start_time}{bt}\n
+    Query Text: {bt}{bt}{bt}{query_text}{bt}{bt}{bt}
+                ';
+               BEGIN
+                   slackResult := (select to_json(INTERNAL.NOTIFICATIONS(tools.templatejs(:message, :dict), 'unused', 'slack', :slackDests)));
+               EXCEPTION
+               WHEN other THEN
+                   slackResult := SQLERRM;
+               END;
+               outcome := outcome || slackResult;
+           END IF;
+
+
+            let name string := act.PROBE_NAME;
+            let action string := act.ACTION_TAKENS;
+            let query_id string := act.QUERY_ID;
+            insert into internal.probe_actions select CURRENT_TIMESTAMP(), :name, :query_id, parse_json(:action), :outcome;
+        END FOR;
+    EXCEPTION
+        WHEN OTHER THEN
+            SYSTEM$LOG_ERROR(OBJECT_CONSTRUCT('error', 'Exception occurred evaluating probes against queries.', 'outcome', :outcome, 'probe_action' :action, 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate));
+            -- TODO this has been broken as it refers to variables that may not be defined. Fix it.
+            insert into internal.probe_actions select CURRENT_TIMESTAMP(), 'unknown name', 'unknown query id', null::variant, :outcome;
+            -- No RAISE to avoid skipping the consumption check
+    END;
+    SYSTEM$LOG_DEBUG('finished probe_monitoring, check quota usage consumption');
+
+    let generate_consumption boolean := (select internal.is_consumption_enabled());
+    if (:generate_consumption) THEN
+        SYSTEM$LOG_INFO('starting usage consumption');
+        let start_time timestamp_ltz := (select current_timestamp());
+        BEGIN
+            call internal.get_daily_quota_select() into :sql;
+            execute immediate sql;
+            let quota_usage object := (select daily_quota from table(result_scan(last_query_id())));
```

Comment:
I found out why we had this pattern for probes -- can't query the information_schema table functions in an owner's rights procedure. But, querying them from a task _is_ ok.